src/Mod/PartDesign/App/FeaturePocket.cpp

```diff
@@ -219,11 +220,20 @@ App::DocumentObjectExecReturn *Pocket::execute(void)
             prism = refineShapeIfActive(prism);
             this->AddSubShape.setValue(prism);
 
-            // Cut the SubShape out of the base feature
-            BRepAlgoAPI_Cut mkCut(base, prism);
-            if (!mkCut.IsDone())
-                return new App::DocumentObjectExecReturn("Pocket: Cut out of base feature failed");
-            TopoDS_Shape result = mkCut.Shape();
+            TopoDS_Shape result;
+            if (!Outside.getValue()) {
+                // Cut the SubShape out of the base feature
+                BRepAlgoAPI_Cut mkCut(base, prism);
+                if (!mkCut.IsDone())
+                    return new App::DocumentObjectExecReturn("Pocket: Cut out of base feature failed");
+                result = mkCut.Shape();

```

Comment:
```suggestion
                result = this->getSolid(mkCut.Shape());
```

Reply:
thanks for the review, pocket originally didn't do this but just used `mkCut.Shape()` should I assume that was just a mistake and it should have used this method all along?  Can you explain what this does?

Reply:
ok it turns out this is also done just a few lines after (238):
`TopoDS_Shape solRes = this->getSolid(result);`
what to do then?
later on result is used again (line 242):
`int solidCount = countSolids(result);`

Reply:
Actually the idea is just to harmonize things. If it's too much work because of existing codebase, then it could stay as is.